### PR TITLE
add plugin id help for config's

### PR DESCRIPTION
**Pull request by romainmenke**
partial fix for : https://github.com/csstools/postcss-preset-env/issues/156

Also see : https://github.com/csstools/postcss-plugins/discussions/86#discussioncomment-1887319

Uses a levenshtein distance to possibly suggest the correct plugin.

```
options: {
	features: {
		"custom-media": true,
		"postcss-logical": true,
		"postcss-logica": true,
	}
},
```

```
Unknown feature: "custom-media" did you mean: "custom-media-queries"
Unknown feature: "postcss-logical" did you mean: "logical-properties-and-values"
Unknown feature: "postcss-logica" did you mean: "logical-properties-and-values"
```

----------

Tests failed because we still had some that were using `color-mod` and this then throw warnings with the new help stuff.

Maybe the entire `insertBefore` `insertAfter` stuff is outdated since this is no longer of PostCSS plugins run? But this is outside the scope of this change.

### CODE REVIEW

File: plugin-packs/postcss-preset-env/src/index.js [hunk 1]
```diff
@@ -113,14 +115,24 @@ const plugin = opts => {
 	const usedPlugins = supportedFeatures.map(feature => feature.plugin);
 	usedPlugins.push(stagedAutoprefixer);
 
+	const internalPlugin = () => {
```
Comment (romainmenke) [inline plugin-packs/postcss-preset-env/src/index.js:118]: This was broken.
It's not possible to have both `OnceExit` and `plugins` in the same PostCSS plugin.

These need to be chained.

I do not know the `exportsTo` plugin so I can't easily add a test for it to double check that everything is fine now.

File: plugin-packs/postcss-preset-env/test/insert.after.expect.css [hunk 2]
```diff
@@ -1,5 +1,3 @@
-$font: system-ui;
-
 .test-variable {
-	font-family: $font;
+	font-family: system-ui, -apple-system, Segoe UI, Roboto, Ubuntu, Cantarell, Noto Sans, sans-serif;
 }
```
Comment (Antonio-Laguna) [inline plugin-packs/postcss-preset-env/test/insert.after.expect.css:3]: This change confuses me a bit. Why is this needed?

**Review by Antonio-Laguna — COMMENTED** [review]
Thanks for this! This is great and I think it'll help in the future to ease the plugin pack usage. Just left a minor comment. I'm sure I'm missing something.

File: plugin-packs/postcss-preset-env/src/lib/plugin-id-help.js [hunk 3]
```diff
@@ -11,6 +11,10 @@ export function pluginIdHelp(featureNamesInOptions, root, result) {
 		const byId = mostSimilar(featureName, featureNames);
 		const byPackage = mostSimilar(featureName, packageNames);
 
+		// TODO :
+		// 1. create markdown docs with the plugin id's
```
Comment (romainmenke) [inline plugin-packs/postcss-preset-env/src/lib/plugin-id-help.js:15]: outside of scope of this change.
But I want to already indicate were in the code this can be mentioned to users.

File: plugin-packs/postcss-preset-env/test/insert.after.expect.css (hunk 2 above)
Comment (romainmenke) [reply plugin-packs/postcss-preset-env/test/insert.after.expect.css:3]: It confused me too.

tape contents for this test :

```js
	'insert:after:array': {
		message: 'supports { stage: 1, after: { "lab-function": [ require("postcss-simple-vars") ] } } usage',
		options: {
			stage: 1,
			insertAfter: {
				'lab-function': [
					require('postcss-simple-vars')()
				]
			},
			features: {
				'lab-function': {
					unresolved: 'ignore'
				}
			}
		},
		expect: 'insert.after.expect.css'
	},
```

I replaced the `color-mod` stuff with `lab-function` to resolve the warnings.
I think that `insertAfter` looks up the index of the plugin (`lab-function`) and tries to add the other plugin after it.

If the feature is not found I think it was never inserted.

So when `color-mod` was removed `postcss-simple-vars` was no longer applied correctly in this test.

Now that I have changed it to `lab-function` it actually works again.
Although probably not as designed because of plugin (not) ordering in PostCSS 8

Comment (Antonio-Laguna) [reply plugin-packs/postcss-preset-env/test/insert.after.expect.css:3]: Hmm we should probably add another issue to test this meaningfully since I don't think this test proves anything right now
